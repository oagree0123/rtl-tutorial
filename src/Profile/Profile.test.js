import React from "react";
import { render, screen } from "@testing-library/react";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile username="oagree" name="오예준" />);
    expect(utils.container).toMatchSnapshot();
  });
  it("show the props correctly", () => {
    render(<Profile username="oagree" name="오예준" />);
    // text를 가진 엘리먼트가 있는지 확인
    screen.getByText("oagree");
    screen.getByText("(오예준)");
    screen.getByText(/오/);
  });
});
