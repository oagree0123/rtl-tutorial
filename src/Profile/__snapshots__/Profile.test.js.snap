// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      oagree
    </b>
     
    <span>
      (
      오예준
      )
    </span>
  </div>
</div>
`;
